### PR TITLE
Automatic dereferencing is removed

### PR DESCRIPTION
**Pull request by ringabout**
implicitDeref is removed

### CODE REVIEW

File: compiler/options.nim [hunk 1]
```diff
@@ -187,7 +187,7 @@ type
     ideHighlight, ideOutline, ideKnown, ideMsg, ideProject
 
   Feature* = enum  ## experimental features; DO NOT RENAME THESE!
-    implicitDeref,
+    implicitDeref, # dead code; remains here for backwards compatibility
```
Comment (metagn) [inline compiler/options.nim]:
```suggestion
    implicitDeref, # removed experimental feature; remains here for backwards compatibility
```

Comment (ringabout) [reply compiler/options.nim]: thanks

File: tests/implicit/timplicit.nim [hunk 2]
```diff
@@ -39,10 +39,10 @@ block:
...
   new(x)
 
-  x.p(44)
+  x[].p(44)
 
   var indirect = p
-  x.indirect(44)
+  x[].indirect(44)
 
   echo x[]
```
Comment (metagn) [inline tests/implicit/timplicit.nim:48]: This entire test is for implicitDeref, except the `sp.a` part. And it occupies its own folder `implicit`.

IMO move lines 18-30 to their own test (maybe `objects/tptrobjectfield.nim`), replace `if sp.a == 2: echo 2` with `doAssert sp.a == 2`, then delete this test and the folder it's in.

**Comment by ringabout** [comment]
Succeeded by https://github.com/nim-lang/Nim/pull/20531

**Pull request closed without merging**